### PR TITLE
fix: Fix hide issue

### PR DESCRIPTION
**Pull request by afonsobspinto**
- Removes double connect 
- Removes timepoint from state (cause of the issue even tho I don't fully get it why) 

https://github.com/MetaCell/geppetto-NeuroSCAN/assets/19196034/7b5de05f-b639-44e9-8933-7c1e902b2032

### CODE REVIEW

**Review by ddelpiano — CHANGES_REQUESTED** [review]

File: applications/neuroscan/frontend/src/components/viewers/viewer.js [hunk 1]
```diff
@@ -244,15 +244,4 @@ class Viewer extends React.Component {
...
 }
-
-const mapDispatchToProps = (dispatch) => ({
-  addInstancesToViewer: (viewerId, instances) => dispatch(
-    addInstances(viewerId, instances, VIEWERS.InstanceViewer),
-  ),
-});
-
-const mapStateToProps = (state, ownProps) => ({
-  timePoint: state.search.filters.timePoint,
```
[the remark this answers is not in the document]
Comment (afonsobspinto) [reply applications/neuroscan/frontend/src/components/viewers/viewer.js:255]: 1. No, that causes the hide issue that this PR is meant to fix
2. In theory no, but after testing I realized that there a few problems with the redux logic meant to update the viewer timepoint which I fixed in the context of this PR.

**Comment by afonsobspinto** [comment]
https://github.com/MetaCell/geppetto-NeuroSCAN/assets/19196034/1e73f768-3b60-4dff-a182-3afd4fa2a154

**Comment by ddelpiano** [comment]
lgtm

**Review by ddelpiano — APPROVED** [review]

File: applications/neuroscan/frontend/src/components/viewers/viewer.js [hunk 2]
```diff
@@ -140,7 +140,7 @@ class Viewer extends React.Component {
 
         const instances = fetchedNeurons.map((neuron) => mapToInstance(neuron));
 
-        addInstancesToViewer(viewerId, instances);
+        addInstancesToViewer(viewerId, instances, timePoint);
```
Comment (zsinnema) [inline applications/neuroscan/frontend/src/components/viewers/viewer.js:143]: question: @afonsobspinto the viewer is connected to a timepoint, why did you add `timePoint` as parameter?

File: applications/neuroscan/frontend/src/redux/middleware.js [hunk 3]
```diff
@@ -43,12 +43,15 @@ import {
...
 const devStagesService = new DevStageService();
 
-const getWidget = (store, viewerId, viewerType) => {
+const getWidget = (store, viewerId, viewerType, timePoint = null) => {
   const state = store.getState();
   const { widgets } = state;
   const widget = widgets[viewerId];
   if (!widget) {
-    const { timePoint } = state.search.filters;
+    if (!timePoint) {
```
Comment (zsinnema) [inline applications/neuroscan/frontend/src/redux/middleware.js:51]: question: @afonsobspinto can't we get the timepoint from the viewer config?

File: applications/neuroscan/frontend/src/redux/middleware.js [hunk 4]
```diff
@@ -90,7 +93,7 @@ const middleware = (store) => (next) => (action) => {
       next(loading(msg, action.type));
       createSimpleInstancesFromInstances(action.instances)
         .then(() => {
-          const widget = getWidget(store, action.viewerId, action.viewerType);
+          const widget = getWidget(store, action.viewerId, action.viewerType, action.timepoint);
```
Comment (zsinnema) [inline applications/neuroscan/frontend/src/redux/middleware.js:96]: @afonsobspinto same here :-)

File: applications/neuroscan/frontend/src/components/viewers/viewer.js (hunk 2 above)
Comment (afonsobspinto) [reply applications/neuroscan/frontend/src/components/viewers/viewer.js:143]: If the widget / viewer exists already the timepoint is not used.

For the situations where the viewer doesn't exist yet, like you said we need to connect it to a timepoint. 

The previous codebase was using the timepoint stored in the redux store to do that. I don't believe that is the correct behavior for situations where I want to add instances to a new viewer from a viewer in a different timepoint than the one currently in store:


https://github.com/MetaCell/geppetto-NeuroSCAN/assets/19196034/1b6c1114-ba5a-4ca5-9736-b348c84776ec

File: applications/neuroscan/backend/api/contact/services/contact.js [hunk 5]
```diff
@@ -75,17 +65,25 @@ module.exports = {
...
-      query.orderBy('neuronA_content.uid', 'asc');
+      query.orderBy('neuronA.uid', 'asc');
+      query.orderBy('neuronB.uid', 'asc');
     }
 
     query.offset(start).limit(limit);
 
-    return query;
+    const ids = await query;
+    return strapi.query('contact').find({ id_in: ids.map(x => x.id), _sort: 'uid' });
```
Comment (afonsobspinto) [inline applications/neuroscan/backend/api/contact/services/contact.js:86]: Isn't the sort by 'uid' breaking the custom logic to sort the contacts?

File: applications/neuroscan/frontend/src/components/viewers/viewer.js [hunk 6]
```diff
@@ -145,18 +145,24 @@ class Viewer extends React.Component {
     }
   }
 
-  handleAddInstancesToViewer = async (viewerId = null) => {
-    const { addInstancesToViewer, timePoint } = this.props;
+  handleCloneViewerWithInstancesList = async (addToViewerId = null) => {
```
Comment (afonsobspinto) [inline applications/neuroscan/frontend/src/components/viewers/viewer.js:148]: I understand the change in the name of this method but given the fact that there's a path where the clone doesn't happen I'm not sure if the renaming was progress.

File: applications/neuroscan/frontend/src/redux/middleware.js [hunk 7]
```diff
@@ -186,34 +218,19 @@ const middleware = (store) => (next) => (action) => {
...
-                    });
-                });
-            });
+          const newNeurons = await fetchDataForEntity(neuronService, timePoint, neurons);
+          const newContacts = await fetchDataForEntity(contactService, timePoint, contacts);
+          const newSynapses = await fetchDataForEntity(synapseService, timePoint, synapses);
+
+          const newInstances = [...newNeurons, ...newContacts, ...newSynapses]
+            .map((i) => mapToInstance(i));
+          widget.config.timePoint = timePoint; // update the current widget's timepoint
```
Comment (afonsobspinto) [inline applications/neuroscan/frontend/src/redux/middleware.js:231]: This line is missing in the CPHATE case:

See https://github.com/MetaCell/geppetto-NeuroSCAN/blob/85f381331e74b012bde7479f0714687752be06b2/applications/neuroscan/frontend/src/redux/middleware.js#L177C1-L178C1

File: applications/neuroscan/backend/api/contact/services/contact.js (hunk 5 above)
Comment (zsinnema) [reply applications/neuroscan/backend/api/contact/services/contact.js:86]: @afonsobspinto nice catch, thanks/ I will change that

File: applications/neuroscan/frontend/src/components/viewers/viewer.js (hunk 6 above)
Comment (zsinnema) [reply applications/neuroscan/frontend/src/components/viewers/viewer.js:148]: agree @afonsobspinto I will revert the rename

File: applications/neuroscan/frontend/src/redux/middleware.js (hunk 7 above)
Comment (zsinnema) [reply applications/neuroscan/frontend/src/redux/middleware.js:231]: nice catch @afonsobspinto , I've added the line

**Comment by zsinnema** [comment]
hi @afonsobspinto can you please re-review? I refactored the custom search

File: applications/neuroscan/backend/api/contact/services/contact.js (hunk 5 above)
Comment (zsinnema) [reply applications/neuroscan/backend/api/contact/services/contact.js]: done

**Comment by afonsobspinto** [comment]
looks good to me :+1:
